File: sol/2099/2099.cpp
#include <ranges>

struct Solution {
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        int n = nums.size();
        vector<int> indices(n);
        ranges::iota(indices, 0);
        ranges::sort(indices, [&](int i, int j){return nums[i] > nums[j];});
        ranges::sort(indices.begin(), indices.begin() + k);
        return views::take(indices, k) 
            | views::transform([&](int i){return nums[i];}) 
            | ranges::to<vector>();
    }
};

File: sol/2099/2099.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        indices = sorted(range(n), key=lambda i: nums[i], reverse=True)[:k]
        indices.sort()
        return [nums[i] for i in indices]
